src/ZoneServer/Skills/Handlers/Swordsman/Hoplite/LongStride.cs

```diff
@@ -0,0 +1,144 @@
+	public class Hoplite_LongStride : IGroundSkillHandler, IDynamicCasted
+	{
+		public const int MaxDistance = 100;
+
+		/// <summary>
+		/// Called when the user starts casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void StartDynamicCast(Skill skill, ICombatEntity caster)
+		{
+		}
+
+		/// <summary>
+		/// Called when the user stops casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void EndDynamicCast(Skill skill, ICombatEntity caster)
+		{
+		}
+
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!skill.Vars.TryGet<Position>("Melia.ToolGroundPos", out var targetPos))
+			{
+				caster.ServerMessage(Localization.Get("No target location specified."));
+				Send.ZC_SKILL_CAST_CANCEL(caster);
+				return;
+			}			
+
+			if (!caster.Map.Ground.IsValidPosition(targetPos))
+			{
+				caster.ServerMessage(Localization.Get("Invalid target location."));
+				Send.ZC_SKILL_CAST_CANCEL(caster);
+				return;
+			}
+
+			if ((int)originPos.Get2DDistance(targetPos) > MaxDistance)
+			{
+				caster.ServerMessage(Localization.Get("Too far."));
+				Send.ZC_SKILL_CAST_CANCEL(caster);
+				return;
+			}
+
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, targetPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, targetPos, null);
+			
+			Send.ZC_NORMAL.LeapJump(caster, targetPos, 0, 0, 0.5f, 0.7f, 0.7f, 90);
+			caster.Position = targetPos;
+
+			this.Attack(skill, caster, new Circle(targetPos.GetRelative(caster.Direction, 15f), 30));
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(800);
+			var damageDelay = TimeSpan.FromMilliseconds(50);
+			var skillHitDelay = TimeSpan.Zero;
+			var afterLandingDelay = TimeSpan.FromMilliseconds(400);
+
+			await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var hits = new List<SkillHitInfo>();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var skillHitResult = SCR_SkillHit(caster, target, skill);
+
+				if (skillHitResult.Result == HitResultType.Crit && caster.TryGetSkill(SkillId.Hoplite_SharpSpear, out var sharpSpear))
```

Comment:
I considered that, but it only applies to specific hoplite skills, so you'd need to check the skill ID in the code.  It could definitely be done using a case statement or some such.